client/components/all-products.js

```diff
@@ -9,21 +9,25 @@ class AllProducts extends Component {
   }
 
   render() {
-    const {products} = this.props.products
-    return (
-      <div>
-        <p>Wands</p>
-        {products.map(product => (
-          <SingleProduct key={product.id} product={product} />
-        ))}
-      </div>
-    )
+    const products = this.props.products
+    if (products) {
+      return (
+        <div>
+          <p>Wands</p>
+          {products.map(product => (
+            <Link to={`single-product/${product.id}`} key={product.id}>
+              {product.name}
+            </Link>
+          ))}
+        </div>
+      )
+    } else return <div>no products</div>
```

Comment:
Do we want to polish up the else block at all? Or just display 'no products', lol

Reply:
Haha maybe for now its ok and later we can switch it to a custom 404 page